src/js/common/schemaform/reducers/index.js

```diff
@@ -21,12 +21,19 @@ function recalculateSchemaAndData(initialState) {
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = _.mapValues('data', state);
```

Comment:
The reason for the pageData/formData separation is mostly because I wasn't sure about doing something like this. All the processing in here happens every time a user changes data in the form. So it makes me nervous. 

I also sort of feel like we should avoid making a lot of dependencies between the different pages, because it creates forms that are hard to understand. This makes it really easy to do that, and makes it harder to tell when you've done so if you're coming back to the code, which also makes me nervous.

I don't have a firm opinion right now, so I'm open to other thoughts on this.